rero_ils/modules/loans/api.py

```diff
@@ -290,6 +290,122 @@ def date_fields2str(self):
             if field in self:
                 self[field] = self[field].isoformat()
 
+    @classmethod
+    def requested_loans_to_validate(cls, library_pid):
+        """Get Requests to be validated."""
+        from ..item_types.api import ItemTypesSearch
+        from ..items.api import ItemsSearch
+
+        # Proxy record
+        patrons = {}
+        locations = {}
+        libraries = {}
+        items = {}
+        item_types = {}
+
+        def loans_pending():
+            """Get pending loans."""
+            return LoansSearch()\
+                .params(preserve_order=True)\
+                .filter('term', state=LoanState.PENDING)\
+                .filter('term', library_pid=library_pid)\
+                .sort({'_created': {"order": 'asc'}})\
+                .source(includes=[
+                    'pid', 'transaction_date', 'item_pid', 'patron_pid',
+                    'document_pid', 'library_pid', 'state',
+                    'transaction_location_pid', 'pickup_location_pid'
+                ])\
+                .scan()
+
+        def patron_by_pid(patron_pid):
```

Comment:
You need to pay attention to memory usage. Because you keep all patrons, items, ... you could load a lot of data and cause memory overflow.

Reply:
Keeping items is useless as we can not have more than one request to validate by item.